### PR TITLE
[RFR] upload aws image and share it with other team accounts.

### PR DESCRIPTION
**Pull request by mshriver**
* upload aws image and share it with other team account.
    addresses# RHCFQE-1296

This branch was completely stolen from @amavinag, taking ownership to replace PR #3848.

I'm coming up to speed on what these are doing and should be doing, but implemented the comments from the previous PR.

### CODE REVIEW

File: scripts/template_upload_ec2.py [hunk 1]
```diff
@@ -224,13 +246,15 @@ def run(**kwargs):
          **kwargs: Kwargs are passed by template_upload_all.
      """
     mgmt_sys = cfme_data['management_systems']
-    for provider in list_provider_keys('ec2'):
+    if 'ec2east' in mgmt_sys:
```
Comment (mshriver) [inline scripts/template_upload_ec2.py]: This change came from the original review - but needs fixing because we need provider set.

**Comment by dajoRH** [comment]
Lint report for commit 66e2e47fee00b63fe28ed6e7c79f26d024c04854:

`scripts/template_upload_ec2.py`:
- :bangbang: Line 188:16: **E121** *continuation line under-indented for hanging indent*
- :red_circle: Line 189:5: **F841** *local variable 'output' is assigned to but never used*
- :heavy_exclamation_mark: Line 249:30: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: scripts/template_upload_ec2.py [hunk 2]
```diff
@@ -224,13 +246,17 @@ def run(**kwargs):
...
      """
     mgmt_sys = cfme_data['management_systems']
-    for provider in list_provider_keys('ec2'):
-        ssh_rhevm_creds = mgmt_sys[provider]['credentials']
-        username = credentials[ssh_rhevm_creds]['username']
-        password = credentials[ssh_rhevm_creds]['password']
-        upload_bucket_name = mgmt_sys[provider]['upload_bucket_name']
-        upload_template(provider, username, password, upload_bucket_name, kwargs.get(
-            'image_url'), kwargs.get('template_name'))
+    valid_providers = ['ec2east']
```
Comment (RonnyPfannschmidt) [inline scripts/template_upload_ec2.py]: from the initial comment it looks like something is missing here?

Comment (mshriver) [inline scripts/template_upload_ec2.py]: @RonnyPfannschmidt I haven't gotten back around to this PR until now - can you elaborate on what you meant here? Looking back at my comment on the old commits, the intent here is to list provider keys here where we want to run and not run on all ec2 keys.

**Comment by mshriver** [comment]
Closing as no longer needed.

**Pull request closed without merging**